uaclient/contract.py

```diff
@@ -492,3 +532,38 @@ def get_contract_information(cfg: UAConfig, token: str) -> Dict[str, Any]:
     """Query contract information for a specific token"""
     client = UAContractClient(cfg)
     return client.request_contract_information(token)
+
+
+def is_contract_changed(cfg: UAConfig) -> bool:
+    orig_token = cfg.machine_token
+    orig_entitlements = cfg.entitlements
+    machine_token = orig_token.get("machineToken", "")
+    contract_id = (
+        orig_token.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("id", "")
+    )
+    contract_client = UAContractClient(cfg)
+    resp = contract_client.get_updated_contract_info(
+        machine_token, contract_id
+    )
+    try:
+        resp_ = json.dumps(resp, cls=util.DatetimeAwareJSONEncoder)
+        resp = json.loads(resp_, cls=util.DatetimeAwareJSONDecoder)
```

Comment:
I think we can parse the date directly using the function from util `parse_rfc3339_date`. This will avoid this encoding/decoding call